Output calls to set_abstract_... in model_init()

Timmmm opened the pull request:
So that you don't need to call these functions automatically. As far as I can tell they are idempotent so it should be backwards compatible.

Commit:
Output calls to set_abstract_... in model_init()

So that you don't need to call these functions automatically. As far as I can tell they are idempotent so it should be backwards compatible.

Timmmm commented:
I confirmed this compiles for sail-riscv. The only thing I wasn't quite sure about is it looks like the function parameters aren't necessarily always `void`? Is it possible that these functions need some input?

Alasdair commented:
Yes, they need input when the abstract type is declared without using a configuration parameter. I think that's what's tripping up the failing test.

Commit:
Don't call `sail_set_abstract_...` when it takes a parameter

If an abstract type hasn't been initialised we can't automatically initialise it.

Timmmm commented:
Ah I see, didn't know you could do that. I've updated it to skip those types. Does feel a bit special-casey but I couldn't think of anything better.

Alasdair commented:
The alternative might be to add parameters to the model_init function we are putting the calls in, but that seems more complicated.
